diff --git a/abseil-cpp/BUILD.gn b/abseil-cpp/BUILD.gn
index d3497e50995..de34e5d733c 100644
--- a/abseil-cpp/BUILD.gn
+++ b/abseil-cpp/BUILD.gn
@@ -114,6 +114,7 @@ config("absl_include_config") {
 
 config("absl_define_config") {
   defines = [ "ABSL_ALLOCATOR_NOTHROW=1" ]
+  defines += [ "_GLIBCXX_USE_CXX11_ABI=0" ]
 
   if (is_win && !use_custom_libcxx) {
     defines += [
diff --git a/jsoncpp/BUILD.gn b/jsoncpp/BUILD.gn
index b51aa3a8a59..fdc1f3fad45 100644
--- a/jsoncpp/BUILD.gn
+++ b/jsoncpp/BUILD.gn
@@ -39,6 +39,7 @@ source_set("jsoncpp") {
   defines = [
     "JSON_USE_EXCEPTION=0",
     "JSON_USE_NULLREF=0",
+    "_GLIBCXX_USE_CXX11_ABI=0",
   ]
 
   include_dirs = [ "source/src/lib_json" ]
