diff --git a/Src/SparseMatrix.inl b/Src/SparseMatrix.inl
index 1111111..2222222 100644
--- a/Src/SparseMatrix.inl
+++ b/Src/SparseMatrix.inl
@@ -463,7 +463,7 @@ namespace Eigen
 
     template<typename T, typename IndexType, typename MatrixRowIterator>
     SparseMatrix< T, IndexType, MatrixRowIterator > operator * (const SparseMatrix< T, IndexType, MatrixRowIterator >& A, const T& s) {
-        for( size_t i=0 ; i<A.rows ; i++ )
+        for( size_t i=0 ; i<A.rows() ; i++ )
         {
             const MatrixRow< T, IndexType >& row = A[i];
             for( size_t j=0 ; j<row.entries ; j++ ) out[i][j].Value *= s;
@@ -496,7 +496,7 @@ namespace Eigen
 
     template< typename T, typename IndexType, typename MatrixRowIterator >
     SparseMatrix< T, IndexType, MatrixRowIterator > operator + ( const SparseMatrix< T, IndexType, MatrixRowIterator >& A, const SparseMatrix< T, IndexType, MatrixRowIterator >& B ) {
-        for( size_t i=0 ; i<A.rows ; i++ )
+        for( size_t i=0 ; i<A.rows() ; i++ )
         {
             const MatrixRow< T, IndexType >& rowA = A[i];
             const MatrixRow< T, IndexType >& rowB = B[i];
