diff --git a/faiss/CMakeLists.txt b/faiss/CMakeLists.txt
index 51680fd..d15d390 100644
--- a/faiss/CMakeLists.txt
+++ b/faiss/CMakeLists.txt
@@ -145,19 +145,14 @@ endif()
 target_compile_definitions(faiss PRIVATE FINTEGER=int)
 
 find_package(OpenMP REQUIRED)
-target_link_libraries(faiss PRIVATE OpenMP::OpenMP_CXX)
-
-find_package(MKL)
-if(MKL_FOUND)
-  target_link_libraries(faiss PRIVATE ${MKL_LIBRARIES})
-else()
-  find_package(BLAS REQUIRED)
-  target_link_libraries(faiss PRIVATE ${BLAS_LIBRARIES})
-
-  find_package(LAPACK REQUIRED)
-  target_link_libraries(faiss PRIVATE ${LAPACK_LIBRARIES})
+if(TARGET OpenMP::OpenMP_CXX)
+    message(STATUS "Building with OpenMP")
+    set(OPENMP_TARGET "OpenMP::OpenMP_CXX")
+    target_link_libraries(faiss PRIVATE ${OPENMP_TARGET})
 endif()
 
+target_link_libraries(faiss PRIVATE ${FAISS_BLAS_LIBRARIES})
+
 install(TARGETS faiss
   EXPORT faiss-targets
   RUNTIME DESTINATION bin
