diff --git a/faiss/CMakeLists.txt b/faiss/CMakeLists.txt
index 51680fd..6fa00ba 100644
--- a/faiss/CMakeLists.txt
+++ b/faiss/CMakeLists.txt
@@ -147,16 +147,7 @@ target_compile_definitions(faiss PRIVATE FINTEGER=int)
 find_package(OpenMP REQUIRED)
 target_link_libraries(faiss PRIVATE OpenMP::OpenMP_CXX)
 
-find_package(MKL)
-if(MKL_FOUND)
-  target_link_libraries(faiss PRIVATE ${MKL_LIBRARIES})
-else()
-  find_package(BLAS REQUIRED)
-  target_link_libraries(faiss PRIVATE ${BLAS_LIBRARIES})
-
-  find_package(LAPACK REQUIRED)
-  target_link_libraries(faiss PRIVATE ${LAPACK_LIBRARIES})
-endif()
+target_link_libraries(faiss PRIVATE ${FAISS_BLAS_TARGET})
 
 install(TARGETS faiss
   EXPORT faiss-targets
