From b1c384482051789ab6ab4decd9f6801338f57ced Mon Sep 17 00:00:00 2001
From: Yixing Lao <yixing.lao@gmail.com>
Date: Sat, 16 Jan 2021 20:28:28 -0800
Subject: [PATCH] allows CXX ABI change with JSONCPP_CXX_ABI

---
 src/lib_json/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lib_json/CMakeLists.txt b/src/lib_json/CMakeLists.txt
index af26476..9b48a4d 100644
--- a/src/lib_json/CMakeLists.txt
+++ b/src/lib_json/CMakeLists.txt
@@ -153,6 +153,7 @@ endif()
 if(BUILD_STATIC_LIBS)
     set(STATIC_LIB ${PROJECT_NAME}_static)
     add_library(${STATIC_LIB} STATIC ${PUBLIC_HEADERS} ${JSONCPP_SOURCES})
+    target_compile_definitions(${STATIC_LIB} PUBLIC _GLIBCXX_USE_CXX11_ABI=${JSONCPP_CXX_ABI})
 
     # avoid name clashes on windows as the shared import lib is alse named jsoncpp.lib
     if(NOT DEFINED STATIC_SUFFIX AND BUILD_SHARED_LIBS)
-- 
2.17.1

